pkg/operator/workloads/api_status.go

```diff
@@ -227,11 +227,7 @@ func apiStatusCode(apiStatus *resource.APIStatus) resource.StatusCode {
 		return resource.StatusStopped
 	}
 
-	if apiStatus.TotalReady() > 0 {
-		return resource.StatusLive
-	}
-
-	if apiStatus.TotalFailed() > 0 {
+	if apiStatus.FailedUpdatedCompute > 0 {
```

Comment:
If an deployment, and it was updated with a new successful deployment, would the api status remain failed until all of the old replicas are deleted?